src/Prima/Pyxis/Form/Autocomplete.elm

```diff
@@ -92,51 +111,161 @@ The configuration of the `Autocomplete`'s `State`.
+initDebouncer secondsDebounce =
+    Debouncer.manual
+        |> Debouncer.settleWhenQuietFor (Just <| Debouncer.fromSeconds secondsDebounce)
+        |> Debouncer.toDebouncer
 
 
 {-| Updates the `Autocomplete`'s `State`.
 -}
-update : Msg -> State -> State
+update : Msg -> State -> ( State, Cmd Msg, Filter )
 update msg ((State state) as stateModel) =
     case msg of
-        OnFilter value ->
-            updateOnFilter (Just value) stateModel
+        OnInput value ->
+            stateModel
+                |> updateOnInput (Just value)
+                |> H.withCmds [ send <| Debounce (Debouncer.provideInput OnFilter) ]
+                |> withoutFilter
+
+        Debounce subMsg ->
+            let
+                ( subModel, subCmd, emittedMsg ) =
+                    Debouncer.update subMsg state.quiteFor100ms
+
+                mappedCmd : Cmd Msg
+                mappedCmd =
+                    Cmd.map Debounce subCmd
+
+                updatedState : State
+                updatedState =
+                    stateModel
+                        |> updateDebouncer subModel
+            in
+            case emittedMsg of
+                Just emitted ->
+                    update emitted updatedState
+                        |> (\( state_, cmd, filter ) -> ( state_, Cmd.batch [ cmd, mappedCmd ], filter ))
+
+                Nothing ->
+                    updatedState
+                        |> H.withCmds [ mappedCmd ]
+                        |> withoutFilter
+
+        OnFilter ->
+            stateModel
+                |> H.withoutCmds
+                |> maybeWithFilter state.filter
 
         OnSelect value ->
             updateOnSelect (Just value) stateModel
+                |> H.withoutCmds
+                |> withoutFilter
 
         OnReset ->
             updateOnReset stateModel
+                |> H.withoutCmds
+                |> withoutFilter
 
         OnKeyPress (Just KeyboardEvents.UpKey) ->
             updateOnKeyUp stateModel
+                |> H.withoutCmds
+                |> withoutFilter
 
         OnKeyPress (Just KeyboardEvents.DownKey) ->
             updateOnKeyDown stateModel
+                |> H.withoutCmds
+                |> withoutFilter
 
         OnKeyPress (Just KeyboardEvents.EnterKey) ->
             updateOnSelect state.focused stateModel
+                |> H.withoutCmds
+                |> withoutFilter
 
         OnKeyPress Nothing ->
             stateModel
+                |> H.withoutCmds
+                |> withoutFilter
+
+
+{-| Internal. Convert `Msg` into `Cmd Msg`, useful to chain updates.
+-}
+send : Msg -> Cmd Msg
+send msg =
+    Task.succeed msg
+        |> Task.perform identity
+
+
+{-| Internal. Append `Filter`, alias of `Maybe String`, to tell parent when update the choices and with which string.
+-}
+maybeWithFilter : Filter -> ( State, Cmd Msg ) -> ( State, Cmd Msg, Filter )
+maybeWithFilter filter ( state, cmd ) =
+    ( state
+    , cmd
+    , filter
```

Comment:
Aggiungo anche che qui si sta facendo un check sul `threshold` per comandare la risposta della funzione, ma quel dato è proprio quello che è stato definito dall'utilizzatore in fase di configurazione dell'autocomplete. 

Inoltre l'utilizzatore può già, attivamente leggere il valore del filtro dell'autocomplete in qualunque momento con `Autocomplete.filterValue myAutocompleteState`
quindi stai semplicemente spostando un calcolo che potrebbe fare il parent qui dentro.

Reply:
non è vero perchè c'è il debounce